src/google.rs

```diff
@@ -0,0 +1,62 @@
+use anyhow::{bail, Result};
+use reqwest::{Client, Response};
+
+use crate::database::Database;
+
+const GOOGLE_PUBLIC_KEY: &str = "https://www.googleapis.com/oauth2/v3/certs";
+const APP_USER_AGENT: &str = concat!(env!("CARGO_PKG_NAME"), "/", env!("CARGO_PKG_VERSION"),);
+
+pub async fn check_key(db: &Database) -> Result<bool> {
+    let result = get_key().await;
+    match result {
+        Ok(response) => {
+            let response_body = response.text().await;
+            let current_key = db.select("google_key");
+            match (response_body, current_key) {
+                (Ok(body), Ok(value)) => {
+                    if body == value {
+                        Ok(true)
+                    } else {
+                        let remove_key = db.delete("google_key");
+                        let insert_key = db.insert("google_key", &body);
```

Comment:
This stores Google's public key into the tree where issues are stored. The public key should be stored in the top level tree (`Database::Db`) instead. The recommended way is to add a dedicated function to `Database` for retrieving and updating Google's public key, and those functions access `Database::Db` internally to modify the top-level tree (by calling [`self.Db.get`](https://docs.rs/sled/latest/sled/struct.Tree.html#method.get) and [`self.Db.insert`](https://docs.rs/sled/latest/sled/struct.Tree.html#method.insert).

Reply:
Got it! I see the difference (instead of self.tree.get use self.Db.get). Will create those functions in database.rs.